website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md

```diff
@@ -0,0 +1,125 @@
+---
+slug: deploy-tabby-to-huggingface-space.md
+title: Deploying a Tabby Instance in Hugging Face Spaces
+authors:
+  - name: Rand Xie
+    title: Contributor of Tabby
+    url: https://github.com/randxie
+    image_url: https://github.com/randxie.png
+  - meng
+tags: [deployment]
+---
+
+[Hugging Face Spaces](https://huggingface.co/spaces) provides an easy way for anyone to host their machine learning models.
+In this blog post, we are going to show you how to deploy a Tabby instance in Hugging Face Spaces. If you have not heard of Tabby, it’s an open-source Github Copilot alternative that supports code completion.
+Check out more details [here](https://github.com/TabbyML/tabby).
+
+### How it works
+
+Let’s firstly take a look at what steps are needed to deploy a Tabby instance in Hugging Face. It’s super easy and you don’t need much coding knowledge. Buckle up and let’s get started.
+
+**Step 1:** Create a Hugging Face account. You will need the account to create your Space. Sign up at [this link](https://huggingface.co/welcome) if you don’t have a Hugging Face account.
+
+**Step 2:** Create a new Hugging Face Space ([link](https://huggingface.co/new-space)). Spaces are code repositories that host application code for Machine Learning demos.
+
+**Step 3:** Create a Dockerfile to capture your machine learning models’ logic, and bring up a server to serve requests.
+
+**Step 4:** After the Dockerfile is built, you will be able to send requests to the APIs.
+
+That’s it! With the hosted APIs, you will be able to call these APIs in your apps. Next, we will deep dive into each step with screenshots!!
+**Everything will be done in the Hugging Face UI. No local setup is needed.**
+
+### Deep Dive
+
+#### Create a new Space
+
+After you create a Hugging Face account, you should be able to see the following page by clicking this [link](https://huggingface.co/new-space).
+The owner name will be your account name. Fill in a Space name, e.g. "tabbyml", and select Docker as Space SDK. Then click "Create Space" at the bottom.
```

Comment:
You might want to mention using "T4" GPU here. By default, newly created space uses 2-Core CPU.